libs/designer/src/lib/core/state/undoRedo/undoRedoTypes.ts

```diff
@@ -19,7 +17,8 @@ import {
 export interface StateHistory {
   past: StateHistoryItem[];
   future: StateHistoryItem[];
-  stateHistoryItemIndex: number;
+  // Toggle whenever undo/redo is clicked. This is needed to re-render parameter panel on undo/redo
```

Comment:
Do we not need the state history item index anymore?

Reply:
No, we just need the key to change when undo/redo is clicked so that parameter panel re-renders. Toggling the key between 0 and 1 works and avoids edge case issues due to us not updating the item index on state save (since we don't want to re-render panel every time we save a state, only when we click undo/redo)